From 7a3d74ab730c29edd582d6cfb25ec794e7b6258b Mon Sep 17 00:00:00 2001
From: Prakash Kumar <prakumar@habana.ai>
Date: Fri, 01 Apr 2022 06:39:15 +0300
Subject: [PATCH] [SW-77158] greco: configure led1 for boot process

This patch configures gpio41 to indicate the booting
stage. At preboot stage led1 should be turned on.

Change-Id: Ib4182f20db8c720fee8855d6d6c690c648a591b4
Signed-off-by: Prakash Kumar <prakumar@habana.ai>
---

diff --git a/arch/greco/arc/asic.c b/arch/greco/arc/asic.c
index 029b7d8..1a7cbcf 100644
--- a/arch/greco/arc/asic.c
+++ b/arch/greco/arc/asic.c
@@ -11,12 +11,20 @@
 #include <pll.h>
 #include <uart.h>
 #include <psoc_pci_pll_ctrl_regs.h>
+#include <psoc_gpio2_regs.h>
+#include <bitops.h>
 #include "common.h"
 
-/* configure yellow LED(GPIO2) in output mode and turn on(active low) */
+/* configure yellow LED(GPIO[41]) in output mode and turn on(active low) */
+#define YELLOW_LED_GPIO		9
 void preboot_led_on(void)
 {
-	/* TODO: perform LED GPIO configuration */
+	u32 val = BIT(YELLOW_LED_GPIO);
+
+	regmap_writel(val, mmPSOC_GPIO2_OUTENSET);
+	val = regmap_readl(mmPSOC_GPIO2_DATAOUT);
+	val &= ~BIT(YELLOW_LED_GPIO);
+	regmap_writel(val, mmPSOC_GPIO2_DATAOUT);
 }
 
 static struct hl_pll_cfg greco_pll_cfg[] = {
